lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -41,6 +43,39 @@ static void EmitNullPropagation(Js::RegSlot targetObjectSlot, ByteCodeGenerator
     );
 }
 
+/// <summary>
+/// The whole optional-chain expression will be wrapped in a UniNode with `knopOptChain`.
+/// Use this function to emit the whole expression.
+/// </summary>
+template <class TEmitProc>
+static void EmitOptionalChainWrapper(ParseNodeUni *pnodeOptChain, ByteCodeGenerator *byteCodeGenerator, FuncInfo *funcInfo, TEmitProc emitChainContent) {
```

Comment:
I'm not sure in how fare the compiler can / will inline this "lambda" here.
Is there a better alternative or is this fine?

Reply:
I'm a little confused what is being accomplished by the lambda here vs directly calling Emit don't you have all the same parameters sitting there?

Also unsure why you pulled this logic out into a function is it called anywhere other than in the Emit Switch statement?

Reply:
This is used in 93698457c0045cb347042beacecaf517dd878ca0 to correctly resolve `this` for a function call on an optional-chain (edge-case)
```js
(some?.optional?.chain?.method)()
```
https://github.com/chakra-core/ChakraCore/blob/93698457c0045cb347042beacecaf517dd878ca0/lib/Runtime/ByteCode/ByteCodeEmitter.cpp#L8115-L8120

Reply:
I almost wonder if it would be neater to get rid of the lambda and just bring the last 3 lines of EmitOptionalChainWrapper to the two places you're calling it.

Reply:
I'm also not a huge fan of the lambda, but it ensures the correct lifetime of `currentOptionalChainSkipLabel` and prevents the function signature from containing 3 out pointers:
 - `innerNode` (plus describing comments)
 - `skipLabel`
 - `previousSkipLabel`

Reply:
Ah - it makes more sense than I'd thought; probably to keep it as is.